internal/connector/connector.go

```diff
@@ -193,15 +192,38 @@ func Run(ctx context.Context, options Options) error {
 		},
 	}
 
+	group, ctx := errgroup.WithContext(ctx)
+
 	con := connector{
 		k8s:         k8s,
 		client:      client,
 		destination: destination,
 		certCache:   certCache,
 		options:     options,
 	}
-	// TODO: make polling time configurable
-	repeat.Start(ctx, 30*time.Second, syncWithServer(con))
+	group.Go(func() error {
+		for {
+			if err := syncWithServer(con); err != nil {
+				return err
+			}
+			wait(ctx, 30*time.Second)
+			if err := ctx.Err(); err != nil {
+				return nil // nolint:nilerr // context cancelled is the shutdown signal
+			}
+		}
+	})
+	group.Go(func() error {
+		for {
+			if err := syncDestination(con); err != nil {
+				logging.Errorf("failed to update destination in infra: %v", err)
+			}
+			// TODO: should this exit after the first successful update?
+			wait(ctx, 30*time.Second)
```

Comment:
I'm wondering what the right behaviour here should be. I think there's a few different cases:

1. the sync failed - we should probably retry pretty quickly, but maybe with some exponential backoff (start at 2s retry, with a max backoff of 1 minute?)
2. the sync was successful, but there's no `Connection.URL` yet (relevant once we merge #3384) - probably the same behaviour as above. Short delay with some backoff
3. the sync was successful, and there is a `Connection.URL` - do we actually want to continue to check for updates in this scenario? I guess it needs to detect new namespaces and roles being added to the cluster? So maybe a 5 minute sleep?

Reply:
The URL could also change. If the service is using NodePort, the original node may have scaled down and replaced with another node. If the service is LoadBalancer, the load balancer may have been updated.